go-controller/pkg/util/ovn.go

```diff
@@ -0,0 +1,54 @@
+package util
+
+// Contains helper functions for OVN
+// Eventually these should all be migrated to go-ovn bindings
+
+import (
+	"fmt"
+	"net"
+	"regexp"
+)
+
+// UpdateRouterNAT checks if a NAT entry exists in OVN for a specific router and updates it if necessary
+// may-exist works only if the the nat rule being added has everything the same i.e.,
+// the type, the router name, external IP and the logical IP must match
+// else the tuple is considered different one than existing.
+// If the type is snat and the logical IP is the same, but external IP is different,
+// even with --may-exist, the add may error out. this is because, for snat,
+// (type, router, logical ip) is considered a key for uniqueness
+// Combining lr-nat-del and lr-nat-add will fail OVSDB transaction check because
+// the entry will already exist so need to do these as two separate transactions
+// Therefore we need to iterate through existing SNAT rules on GR and decide if we need to delete/add
+// TODO(trozet) use go-bindings
+func UpdateRouterNAT(router, natType string, externalIP net.IP, logicalSubnet *net.IPNet) error {
+	ovnNatTypes := map[string]struct{}{"snat": {}, "dnat": {}, "dnat_and_snat": {}}
+	if _, ok := ovnNatTypes[natType]; !ok {
+		return fmt.Errorf("invalid NAT type specified: %s", natType)
+	}
+	stdout, stderr, err := RunOVNNbctl("lr-nat-list", router)
+	if err != nil {
+		return fmt.Errorf("failed to get NAT rules on gateway router %s, "+
+			"stdout: %q, stderr: %q, error: %v", router, stdout, stderr, err)
+	}
+	logicalIPVal := logicalSubnet.String()
+	logicalIPMask, _ := logicalSubnet.Mask.Size()
+	// Values with full prefix mask wont print the /mask
```

Comment:
it does print the mask for me with full mask :thinking: https://play.golang.org/p/TXiGiNWzrzp

Reply:
I should have been more specific here. I meant the output in OVN wont print with the mask if it is full mask.

Reply:
ok :+1: